Add `--mlir-only` option to `run-suite` script

victor-eds opened the pull request:
Add option to run only SYCL-MLIR benchmarks, i.e., those under `pattern` and `polybench`.

whitneywhtsang reviewed (commented):
Although we only want to run performance for single-kernel and polybench for sycl-mlir, do we want to ensure the other benchmarks are functionally clean?

victor-eds commented:
> Although we only want to run performance for single-kernel and polybench for sycl-mlir, do we want to ensure the other benchmarks are functionally clean?

That is a good point. I'll keep this on hold just in case we need it, but will install everything for now in CI.

Commit:
Add `--mlir-only` option to `run-suite` script

Add option to run only SYCL-MLIR benchmarks, i.e., those under
`pattern` and `polybench`.

Signed-off-by: Victor Perez <victor.perez@codeplay.com>

victor-eds commented:
I opted to add a `--mlir-only` option to the `run-suite` script in the end. This will allow us to build all benchmarks and run only those we care bout.

etiotto commented:
> I opted to add a `--mlir-only` option to the `run-suite` script in the end. This will allow us to build all benchmarks and run only those we care bout.

That is OK although we have to remember to run all the one that build cleanly from time to time.
